LobbyServer2/LobbyServer/LobbyServerProtocol.cs

```diff
@@ -678,8 +730,22 @@ public void HandleGroupConfirmationResponse(GroupConfirmationResponse response)
                     log.Info($"Player {AccountId} accepted request {response.ConfirmationNumber} " +
                              $"to join group {response.GroupId} by {response.JoinerAccountId}: {response.Acceptance}");
                     // TODO validation
-                    GroupManager.JoinGroup(response.GroupId, AccountId);
-                    BroadcastRefreshFriendList();
+                    GroupInfo group = GroupManager.GetGroup(response.GroupId);
+                    if (group.Members.Count < 5)
+                    {
+                        GroupManager.JoinGroup(response.GroupId, AccountId);
+                        BroadcastRefreshFriendList();
+                    } 
```

Comment:
This check should happen in GroupManager under lock

Also we probably should not allow sending invites when the group is full